examples/04.temporal_safety/README.md

```diff
@@ -43,3 +45,24 @@ Any of these can be cleared from a copy of the pointer before passing it to othe
 For example, if you remove `W` and `M` permissions from a pointer that you pass as a parameter then you have a guarantee that nothing reachable from the pointer will be mutated.
 Similarly, if you remove `G` and `L` then you have the guarantee that nothing reachable from the pointer will be captured.
 If you remove `G` but not `L` then you have the weaker guarantee that the pointer that you passed will not be captured but pointers reachable from it might be.
+
+The next three use cases show the handling of sub object, a capability that references a sub-range of the allocation.
+
+In the first of these the sub object is passed to free(), but this has no impact until the original allocation is also freed (i.e an allocation does not become fragmented if you try to free the middle of it).
+
+In the second use case a claim is made on the sub object, which counts against the quota but now means that the sub object remains valid even if the enclosing allocation is freed.
+Both the sub object and the enclosing allocation become invalid when the claim is releases. 
+
+The third use case shows the difference between a claim and a fast claim.
```

Comment:
This might be good motivation to finally rename "fast" claims, perhaps before landing an updated version of this PR?  @davidchisnall?